opensearch-operator/pkg/builders/dashboards.go

```diff
@@ -52,10 +52,10 @@ func NewDashboardsDeploymentForCR(cr *opsterv1.OpenSearchCluster, volumes []core
 		env = append(env, cr.Spec.Dashboards.Env...)
 	}
 
-	if cr.Spec.Dashboards.OpensearchCredentialsSecret.Name != "" {
+	if cr.Spec.Security.Config.AdminCredentialsSecret.Name != "" {
```

Comment:
IMO This logic should be in addition to the existing one, not replace it.
So 3 cases:
* Dashboards.OpensearchCredentialsSecret is defined -> use it
* Security.Config.AdminCredentialsSecret is defined -> use that instead
* Neither is defined -> use default admin/admin